components/WorkoutsList/index.js

```diff
@@ -2,15 +2,61 @@ import Link from "next/link";
 import WorkoutForm from "../WorkoutForm";
 import WorkoutPreview from "../WorkoutPreview";
 import styled from "styled-components";
+import ModalDelete from "../ModalDelete";
+import { useState } from "react";
+
+export default function WorkoutsList({
+  onAddWorkout,
+  onDeleteWorkout,
+  workouts,
+  exercises,
+}) {
+  const [modalDelete, setModalDelete] = useState(false);
+  const [workoutIdToDelete, setWorkoutIdToDelete] = useState(0);
+  const [isWorkoutDeleted, setIsWorkoutDeleted] = useState(false);
+
+  const handleDelete = (event) => {
+    const id = event.target.dataset.workoutid; // data-workoutid auslesen
+    setWorkoutIdToDelete(id); // id speichern
+    setModalDelete(true);
+  };
+
+  function handleDeleteCancel() {
+    setModalDelete(false);
+  }
+
+  function handleDeleteConfirm() {
+    onDeleteWorkout(workoutIdToDelete);
+    setModalDelete(false);
+    setIsWorkoutDeleted(true);
+  }
 
-export default function WorkoutsList({ onAddWorkout, workouts, exercises }) {
   return (
     <>
+      {workouts.length != 0 && isWorkoutDeleted && (
+        <SuccessMessage>Workout deleted successfully!</SuccessMessage>
+      )}
+
+      {workouts.length == 0 && (
+        <SuccessMessage>
+          Oops! No Workouts yet.<br></br>
+          Create a new Workout to start your journey!
+        </SuccessMessage>
+      )}
+
       <WorkoutCard>
         <WorkoutForm exercises={exercises} onAddWorkout={onAddWorkout} />
         {workouts.map((workout) => (
           <WorkoutItem key={workout.id}>
-            <EditLink href={`workouts/${workout.id}/edit`}>Edit 🖊</EditLink>
+            <Actions>
+              <DeleteButton data-workoutid={workout.id} onClick={handleDelete}>
+                ✘ Delete 
+              </DeleteButton>
+              <EditLink href={`workouts/${workout.id}/edit`}>
+                ✎ Edit 
+              </EditLink>
+            </Actions>
```

Comment:
You should consider having a 'workout' component that renders a workout. This one should also render the buttons. It will receive 'workout' and 'handleDelete' as props and can call the delete function onDelete(workout.id).